tunder/lib/src/database/operations/contracts/database_operator.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:tunder/database.dart';
+import 'package:tunder/src/database/operations/postgres/postgres_operator.dart';
+import 'package:tunder/utils.dart';
+
+abstract class DatabaseOperator {
+  factory DatabaseOperator.forDriver(Symbol driver) {
+    if (driver == #postgres) return app(PostgresDatabaseOperator);
+
+    throw UnsupportedError('Driver [${driver.name}] is not supported');
+  }
+
+  Future insert(MappedRow row, {required String table});
+  Future<int> count(Query query);
+  Future<int> delete(Query query);
+  Future<int> drop(String table);
+  Future<int> dropIfExists(String table);
+  Future<int> dropAllTables();
+  Future<int> update(Query query, MappedRow row);
+  Future<List<MappedRow>> process(Query query);
+  String toSql(Query query);
```

Comment:
@Wdestroier I followed your suggestion about refactoring the database operations to put them all in a class that makes it easier to implement for other drivers and here you are. Now, we just need to implement two classes `DatabaseOperator` and `SchemaProcessor` to implement a new database driver. In the future, I will also refactor the SchemaProcessor to use the database operator instead. This will make it, even more, easier to implement in the future. Please let me know if you have any other thoughts or suggestions. 

Now I just implemented the `PostgresDatabaseOperator` to follow the database operator contract (interface). The `postgres` directory contains everything related to the Postgres driver implementation. It's a good pattern but we don't necessarily need to follow it, but I think it's a good way to implement this contract.